process/dataPreProcessor.go

```diff
@@ -0,0 +1,186 @@
+func NewEventsDataPreProcessor(args ArgsEventsDataPreProcessor) (*eventsDataPreProcessor, error) {
+	err := checkDataIndexerArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	di := &eventsDataPreProcessor{
+		marshaller:         args.Marshaller,
+		internalMarshaller: args.InternalMarshaller,
+		facade:             args.Facade,
+	}
+
+	emptyBlockContainer, err := createEmptyBlockCreatorContainer()
+	if err != nil {
+		return nil, err
+	}
+
+	di.emptyBlockCreator = emptyBlockContainer
+
+	return di, nil
+}
+
+func checkDataIndexerArgs(args ArgsEventsDataPreProcessor) error {
+	if check.IfNil(args.Marshaller) {
+		return common.ErrNilMarshaller
+	}
+	if check.IfNil(args.InternalMarshaller) {
+		return fmt.Errorf("%w (internal)", common.ErrNilMarshaller)
+	}
+	if check.IfNil(args.Facade) {
+		return common.ErrNilFacadeHandler
+	}
+
+	return nil
+}
+
+// SaveBlock will handle the block info data
+func (d *eventsDataPreProcessor) SaveBlock(outportBlock *outport.OutportBlock) error {
+	err := checkBlockDataValid(outportBlock)
+	if err != nil {
+		return err
+	}
+
+	header, err := d.getHeaderFromBytes(core.HeaderType(outportBlock.BlockData.HeaderType), outportBlock.BlockData.HeaderBytes)
+	if err != nil {
+		return err
+	}
+
+	saveBlockData := &data.ArgsSaveBlockData{
+		HeaderHash:             outportBlock.BlockData.HeaderHash,
+		Body:                   outportBlock.BlockData.Body,
+		SignersIndexes:         outportBlock.SignersIndexes,
+		NotarizedHeadersHashes: outportBlock.NotarizedHeadersHashes,
+		HeaderGasConsumption:   outportBlock.HeaderGasConsumption,
+		AlteredAccounts:        outportBlock.AlteredAccounts,
+		NumberOfShards:         outportBlock.NumberOfShards,
+		TransactionsPool:       outportBlock.TransactionPool,
+		Header:                 header,
+	}
+
+	// TODO: refactor to remove facade versioning
+	err = d.facade.HandlePushEventsV2(*saveBlockData)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *eventsDataPreProcessor) getHeaderFromBytes(headerType core.HeaderType, headerBytes []byte) (header coreData.HeaderHandler, err error) {
+	creator, err := d.emptyBlockCreator.Get(headerType)
+	if err != nil {
+		return nil, err
+	}
+
+	return block.GetHeaderFromBytes(d.internalMarshaller, creator, headerBytes)
+}
+
+func checkBlockDataValid(block *outport.OutportBlock) error {
+	if block.BlockData == nil {
+		return ErrNilBlockData
+	}
+	if block.TransactionPool == nil {
+		return ErrNilTransactionPool
+	}
+	if block.HeaderGasConsumption == nil {
+		return ErrNilHeaderGasConsumption
+	}
+
+	return nil
+}
+
+// RevertIndexedBlock will handle the revert block event
+func (d *eventsDataPreProcessor) RevertIndexedBlock(blockData *outport.BlockData) error {
+	header, err := d.getHeaderFromBytes(core.HeaderType(blockData.HeaderType), blockData.HeaderBytes)
+	if err != nil {
+		return err
+	}
+
+	revertData := &data.RevertBlock{
+		Hash:  string(header.GetRootHash()),
```

Comment:
do you use string of bytes everywhere? We had issues in the past when using string(bytes) vs hex encode(bytes)